File: scraper-service/models/user_task.py
from datetime import datetime
from utils import db


class UserTask(db.Model):
    __tablename__ = 'user_tasks'

    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(80), nullable=False)
    task_id = db.Column(db.String(36), nullable=False)
    start_date = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
    end_date = db.Column(db.DateTime)

    def __repr__(self):
        return f'<UserTask {self.username} - {self.task_id}>'

File: web-api/src/main/java/com/amazoninsight/llmreview/model/UserTask.java
package com.amazoninsight.llmreview.model;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;


@Entity
@Table(name = "user_tasks")
@Data
public class UserTask {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "username", nullable = false)
    private String username;

    @Column(name = "task_id", nullable = false)
    private String taskId;

    @Column(name = "start_date", nullable = false)
    private LocalDateTime startDate;

    @Column(name = "end_date")
    private LocalDateTime endDate;

    @Column(name = "asin", nullable = false)
    private String asin;

}